tests/utils/test_autolog_hparams.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+
+from composer.algorithms import EMA
+from composer.callbacks import SpeedMonitor
+from composer.loggers import MLFlowLogger
+from composer.trainer import Trainer
+from composer.utils import convert_flat_dict_to_nested_dict, convert_nested_dict_to_flat_dict, extract_hparams
+from tests.common.datasets import RandomClassificationDataset
+from tests.common.models import SimpleModel
+
+
+def test_convert_nested_dict_to_flat_dict():
+    test_nested_dict = {'a': 1, 'b': {'c': 2, 'd': 3}, 'e': {'f': {'g': 4}}}
+
+    expected_flat_dict = {'a': 1, 'b/c': 2, 'b/d': 3, 'e/f/g': 4}
+    actual_flat_dict = convert_nested_dict_to_flat_dict(test_nested_dict)
+    assert actual_flat_dict == expected_flat_dict
+
+
+def test_convert_flat_dict_to_nested_dict():
+    expected_nested_dict = {'a': 1, 'b': {'c': 2, 'd': 3}, 'e': {'f': {'g': 4}}}
+
+    test_flat_dict = {'a': 1, 'b/c': 2, 'b/d': 3, 'e/f/g': 4}
+    actual_nested_dict = convert_flat_dict_to_nested_dict(test_flat_dict)
+    assert actual_nested_dict == expected_nested_dict
+
+
+def test_extract_hparams():
+
+    class Foo:
+
+        def __init__(self):
+            self.g = 7
+
+    class Bar:
+
+        def __init__(self):
+            self.local_hparams = {'m': 11}
+
+    locals_dict = {'a': 1.5, 'b': {'c': 2.5, 'd': 3}, 'e': [4, 5, 6.2], 'f': Foo(), 'p': Bar()}
+
+    expected_parsed_dict = {
+        'a': 1.5,
+        'b': {
+            'c': 2.5,
+            'd': 3
+        },
+        'e': [4, 5, 6.2],
+        'f': 'Foo',
+        'p': {
+            'Bar': {
+                'm': 11,
+            }
+        },
+    }
+
+    parsed_dict = extract_hparams(locals_dict)
+    assert parsed_dict == expected_parsed_dict
+
+
+def test_extract_hparams_trainer():
+    train_dl = DataLoader(RandomClassificationDataset(), batch_size=16)
+    model = SimpleModel()
+    optimizer = Adam(model.parameters(), eps=1e-3)
+    trainer = Trainer(
+        model=model,
+        train_dataloader=train_dl,
+        optimizers=optimizer,
+        auto_log_hparams=True,
+        progress_bar=False,
+        log_to_console=False,
+        run_name='test',
+        seed=3,
+        algorithms=[EMA()],
+        loggers=[MLFlowLogger()],
```

Comment:
Maybe do `InMemoryLogger`, otherwise this tests needs a conditional check if mlflow is installed.

Reply:
fixed